x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlCCSUtils.java

```diff
@@ -190,16 +198,17 @@ static void updateExecutionInfoWithClustersWithNoMatchingIndices(
         IndexResolution indexResolution,
         QueryBuilder filter
     ) {
-        Set<String> clustersWithResolvedIndices = new HashSet<>();
-        // determine missing clusters
+        final Set<String> clustersWithNoMatchingIndices = new HashSet<>(executionInfo.clusterAliases());
         for (String indexName : indexResolution.resolvedIndices()) {
-            clustersWithResolvedIndices.add(RemoteClusterAware.parseClusterAlias(indexName));
+            clustersWithNoMatchingIndices.remove(RemoteClusterAware.parseClusterAlias(indexName));
         }
-        Set<String> clustersRequested = executionInfo.clusterAliases();
-        Set<String> clustersWithNoMatchingIndices = Sets.difference(clustersRequested, clustersWithResolvedIndices);
-        clustersWithNoMatchingIndices.removeAll(indexResolution.unavailableClusters().keySet());
-
-        /**
+        clustersWithNoMatchingIndices.removeAll(EsqlCCSUtils.determineUnavailableRemoteClusters(indexResolution.failures()).keySet());
```

Comment:
I am not sure this logic is the best way. If the cluster is unavailable here, we should either skip it or error out, in either case it shouldn't be in the list of active clusters. So maybe it should be like this:

1. Start with the list of currently active clusters (not all aliases since there's no point in checking clusters that were skipped e.g. by enrich resolution)
2. Exclude all clusters that became unavailable due to `indexResolution.failures()` - this would skip those which we can skip, or throw I presume if we can't.
3. Now take the list of _still_ active clusters and check the indices for those.